### ISSUE TITLE
Installer not checking password length

### ISSUE THREAD

**Issue by Ben-MyBB** [mybb/mybb#3620]
When installing a fresh copy of MyBB you are able to set a password length of 1 character but when signing up on the board you are required a minimum of 6 characters.

It seems the length is not being checked in the installer.

Original thread: [Installer not checking password length](https://community.mybb.com/thread-222341.html)

**Comment by dvz** [comment]
IMO it shouldn't be as strict in the installer and the ACP - makes test/development board management more difficult.

**Comment by Eldenroot** [comment]
Ok, but it should require at least 6 characters, we are in 21st century, passwords should be secure

**Comment by WildcardSearch** [comment]
@Eldenroot I agree with you about password strength for accounts created through traditional registration, but @dvz has a point that accounts created through the installer/acp are being created by the owner/admin of the forum and they should be able to choose (based on whether it is a test board or live forum) which password is appropriate. jmo

**Comment by chack1172** [comment]
I agree with @dvz, I install many test board in my local pc and every time I set password 'a'. I don't think we should force 6 characters in installer.

**Comment by veryard** [comment]
I agree with @Eldenroot, 6 character minimum, i'm sure we could add an exception if debug is enabled in config.

**Comment by AndreRl** [comment]
IMO I think we need to consider the purpose of this stage in the installer. Is it to make setting up a test environment easier and quicker or to setup an admin account for a forum that's likely to be public? 

We should emphasise the importance of security.

**Comment by WildcardSearch** [comment]
> Is it to make setting up a test environment easier

Just to clarify, my point isn't about making things easier on developers.

I am stating that the choice should be left to the owner/admin. User password strength requirements are imposed *by the owner/admin* and that's why they make sense. We should not impose requirements that remove possibilities. Quite a few changes in the last few releases have made seemingly innocuous changes that have generated a lot of negative feedback. This would be another one, imo.

**Comment by veryard** [comment]
If you think that's the case, why not add a help line below the password
field and say something along the lines of "We recommend that you provide a
strong password etc etc."

Brad.

**Comment by arcalys** [comment]
What should be done about this?

**Comment by dvz** [comment]
The 1.9 installer shows a strength meter with a warning, so both concerns should be addressed.
